opentelemetry-sdk/src/export/logs/mod.rs

```diff
@@ -13,8 +13,9 @@ use std::fmt::Debug;
 /// `LogExporter` defines the interface that log exporters should implement.
 #[async_trait]
 pub trait LogExporter: Send + Sync + Debug {
-    /// Exports a batch of [`LogData`].
-    async fn export(&mut self, batch: Vec<LogData>) -> LogResult<()>;
+    /// Exports a batch of reference to `LogData`.
+    async fn export<'a>(&mut self, batch: &'a [&'a LogData]) -> LogResult<()>;
```

Comment:
is the exported type a slice of ref to LogData? Previously, it was a concrete type `Vec`. Please call it out in changelog. 
(I agree with this change, just making sure the change is aptly reflected in changelog)

Reply:
Had to revert the slice changes as they complicated the handling in the batch processor. Previously, the batch processor directly moved the logs from the queue to the export method. Now, we pass a reference to the logs, track which logs have been passed, and then delete them from the queue. Since the queue can receive new logs while the export is happening, it is necessary to keep track of the logs that were exported. Also, didn't see any improvement in stress and batch tests with the changes.